ts/src/pro/bitpanda.ts

```diff
@@ -0,0 +1,1360 @@
+    describe () {
+        return this.deepExtend (super.describe (), {
+            'has': {
+                'ws': true,
+                'watchBalance': true,
+                'watchTicker': true,
+                'watchTickers': true,
+                'watchTrades': false,
+                'watchMyTrades': true,
+                'watchOrders': true,
+                'watchOrderBook': true,
+                'watchOHLCV': true,
+            },
+            'urls': {
+                'test': {
+                    'ws': 'wss://test.bitpanda.com/ws/api/v2',
+                },
+                'api': {
+                    'ws': 'wss://streams.exchange.bitpanda.com',
+                },
+            },
+            'options': {
+                'bp_remaining_quota': 200,
+                'timeframes': {
+                    '1m': {
+                        'unit': 'MINUTES',
+                        'period': 1,
+                    },
+                    '5m': {
+                        'unit': 'MINUTES',
+                        'period': 5,
+                    },
+                    '15m': {
+                        'unit': 'MINUTES',
+                        'period': 15,
+                    },
+                    '30m': {
+                        'unit': 'MINUTES',
+                        'period': 30,
+                    },
+                    '1h': {
+                        'unit': 'HOURS',
+                        'period': 1,
+                    },
+                    '4h': {
+                        'unit': 'HOURS',
+                        'period': 4,
+                    },
+                    '1d': {
+                        'unit': 'DAYS',
+                        'period': 1,
+                    },
+                    '1w': {
+                        'unit': 'WEEKS',
+                        'period': 1,
+                    },
+                    '1M': {
+                        'unit': 'MONTHS',
+                        'period': 1,
+                    },
+                },
+            },
+            'streaming': {
+            },
+            'exceptions': {
+            },
+        });
+    }
+
+    async watchBalance (params = {}) {
+        /**
+         * @method
+         * @name bitpanda#watchBalance
+         * @see https://developers.bitpanda.com/exchange/#account-history-channel
+         * @description query for balance and get the amount of funds available for trading or funds locked in orders
+         * @param {object} params extra parameters specific to the bitpanda api endpoint
+         * @returns {object} a [balance structure]{@link https://docs.ccxt.com/en/latest/manual.html?#balance-structure}
+         */
+        await this.authenticate (params);
+        const url = this.urls['api']['ws'];
+        const messageHash = 'balance';
+        const subscribeHash = 'ACCOUNT_HISTORY';
+        const bpRemainingQuota = this.safeNumber (this.options, 'bp_remaining_quota', 200);
```

Comment:
`safeNumber` might potentially return a string, so we should use `safeInteger` instead (remaining requests seems to be an integer)